modules/log4cats/src/main/scala/org/legogroup/woof/log4cats/package.scala

```diff
@@ -0,0 +1,72 @@
+package org.legogroup.woof.log4cats
+
+import cats.Applicative
+import org.legogroup.woof
+import org.legogroup.woof.Logger.withLogContext
+import org.typelevel.log4cats
+
+def WoofFactory[F[_]: Applicative]: Conversion[woof.Logger[F], log4cats.LoggerFactory[F]] =
+  new Conversion[woof.Logger[F], log4cats.LoggerFactory[F]]:
+
+    def apply(logger: woof.Logger[F]): log4cats.LoggerFactory[F] =
+      new log4cats.LoggerFactory[F]:
+
+        override def fromName(name: String): F[log4cats.SelfAwareStructuredLogger[F]] =
+          Applicative[F].pure(getLoggerFromName(name))
+
+        override def getLoggerFromName(name: String): log4cats.SelfAwareStructuredLogger[F] =
+          new log4cats.SelfAwareStructuredLogger[F]:
+
+            private def logInfo(): woof.LogInfo =
+              val stacktraceElements = (new Throwable).getStackTrace()
+              val lastIndex = stacktraceElements.reverse.indexWhere(s => s.getClassName == this.getClass.getName)
+              val callingMethodIndex = stacktraceElements.size - lastIndex
+              val callingMethod      = stacktraceElements(callingMethodIndex)
+              val fileName =
+                (callingMethod.getClassName.replace('.', '/') + ".scala").split("\\/").takeRight(1).mkString
```

Comment:
On JS it returns something like `(<jscode>.scala:96229)`.

Reply:
[`log4cats` doesn't support sourcecode info](https://github.com/typelevel/log4cats/issues/525), but it's a requirement for `woof`.

Perhaps, I could create a variant of [`Logger#doLog`](https://github.com/LEGO/woof/blob/ab1a96fba948661dd50fc10a84219254d688c6b8/modules/core/shared/src/main/scala/org/legogroup/woof/Logger.scala#L21) which doesn't require `LogInfo` to facilitate `log4cats` integration.